vendor/github.com/google/cel-go/parser/parser.go

```diff
@@ -352,6 +353,13 @@ func (p *parser) parse(expr runes.Buffer, desc string) *exprpb.Expr {
 
 // Visitor implementations.
 func (p *parser) Visit(tree antlr.ParseTree) interface{} {
+	p.recursionDepth++
+	if p.recursionDepth > p.maxRecursionDepth {
+		panic(&recursionError{message: "max recursion depth exceeded"})
```

Comment:
does this panic get hit if we exceed the max depth while parsing a CRD validation rule?

if so, does it get caught and treated as an error correctly?

Reply:
The `panic` is recovered and turned into a parse error with a reasonable message.